graphx/src/main/scala/org/apache/spark/graphx/util/PeriodicGraphCheckpointer.scala

```diff
@@ -87,10 +87,10 @@ private[mllib] class PeriodicGraphCheckpointer[VD, ED](
 
   override protected def persist(data: Graph[VD, ED]): Unit = {
     if (data.vertices.getStorageLevel == StorageLevel.NONE) {
-      data.vertices.persist()
```

Comment:
We need to use `cache` because `persist` does not honor the default storage level requested when constructing the graph. Only `cache` does that. It's confusing, but true. To verify this for yourself, change these values to `persist` and run the `PeriodicGraphCheckpointerSuite` tests.